Assignment02/main.js

```diff
@@ -0,0 +1,195 @@
+                errorFlag = true; //setting boolean value to true to prevent default and create errorDiv in case of blank value
+            }
+        } else if(mealForm.value === "dinner"){ //if statement to check if dinner meal was selected
+            if(mealOptions[6].checked === false && mealOptions[7].checked === false && mealOptions[8].checked === false){ // if statement to check if any dinner options were checked
+                createListItem("Dinner choice must be selected"); //calling method to create li for errorDiv
+                errorFlag = true; //setting boolean value to true to prevent default and create errorDiv in case of blank value
+            }
+        }
+        if(checkValue(rating.value)){ //checkValue method called to validate input of meal rating
+            createListItem("You must select a meal rating"); //calling method to create li for errorDiv
+            errorFlag = true; //setting boolean value to true to prevent default and create errorDiv in case of blank value
+        }
+        if(errorFlag == true){ //if statement checking whether errorFlag bool value was changed to true
+            event.preventDefault(); //preventing form submit
+            errorHeader.innerHTML = "The following errors must be fixed prior to submitting:"; //adding error message to header
+            errorDiv.appendChild(errorHeader); //appending errorHeader to errorDiv
+            errorDiv.appendChild(errorUL); //appending errUL to errorDiv
+            errorDiv.classList.add("errorBox"); //adding errorBox class to errorDiv
+            errorDiv.classList.add("visible"); //adding visible class to errorDiv
+        }
+    });
+    
+    mealForm.addEventListener("change", (e) => { //added event listener for change event on meal selection
+        makeHidden(mealOptionsArray); //setting meal choices for each respective meal selection to hidden
+        mealOptionsHolder.setAttribute("class", "visible"); //setting meal options div to visible
+        switch(mealForm.value){ //switch statement for mealForm value
+            case "breakfast": //if case is breakfast
+                        breakfast.setAttribute("class", "visible"); //setting breakfast options to visible
+                        break; //break out of switch
+            case "lunch": //if case is lunch
+                        lunch.setAttribute("class", "visible"); //setting lunch options to visible
+                        break; //break out of switch
+            case "dinner": //if case is dinner
+                        dinner.setAttribute("class", "visible"); //setting dinner options to visible
+                        break; //break out of switch
+            default: // default case for the options
+                        mealOptionsHolder.setAttribute("class", "hidden"); //div is set to hidden
+                        break; //break out of switch
+        }   
```

Comment:
This works well because we are able to overwrite the full class list of the input with each change. But....what if we couldn't? What if we need a class for mobile responsiveness AND a visible/hidden? How might your code change? Just something to think about  - great switch statement 👍